### PR TITLE
ENH: Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by kaspermarstal**

### CODE REVIEW

**Comment by N-Dekker** [comment]
Very much looking forward to this feature, @kaspermarstal   `:-)`

**Comment by N-Dekker** [comment]
By the way, when you're just doing trial-and-error to get such a configuration right, I think it would be OK to do `git commit --amend`, followed by `git push --force`. (In general, `push --force` is a bit risky, but as long as you do it on your own feature branch, and you're the only one on this branch, I think it's OK.)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- job: Windows
+  timeoutInMinutes: 0
+  pool:
+    vmImage: 'vs2017-win2016'
+  strategy:
+    matrix:
+      ITK4:
+        itk.version: 'v4.13.1'
+      ITK5:
+        itk.version: 'v5.0b03'
```
Comment (N-Dekker) [inline azure-pipelines.yml]: In general I agree that it's nice to use tagged ITK versions. But unfortunately, with both of these two (v4.13.1 and v5.0b03), elastix still suffers from https://github.com/InsightSoftwareConsortium/ITK/issues/350 Would it be possible to pick the last versions from https://github.com/InsightSoftwareConsortium/ITK/tree/release-4.13 and ITK git master instead?

Comment (kaspermarstal) [reply azure-pipelines.yml]: Yes sure, done!

**Comment by kaspermarstal** [comment]
> By the way, when you're just doing trial-and-error to get such a configuration right, I think it would be OK to do `git commit --amend`, followed by `git push --force`. (In general, `push --force` is a bit risky, but as long as you do it on your own feature branch, and you're the only one on this branch, I think it's OK.)

Yes, I will squash and merge once it is finished.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+variables:
+  PreferredToolArchitecture: x64
```
Comment (N-Dekker) [inline azure-pipelines.yml]: Nitpick:  `PreferredToolArchitecture` is unused. I guess you might as well remove it.

Comment (kaspermarstal) [reply azure-pipelines.yml]: Are you sure it is unused? How do you know?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+- job: macOS
+  timeoutInMinutes: 0
+  pool:
+    vmImage: 'macOS-10.13'
+  strategy:
+    matrix:
+      ITK4:
+        itk.version: 'release-4.13'
+      ITK5:
+        itk.version: 'e018653622038ca7651d21769acfec539f01344d' # master branch as of 22-01-2019
```
Comment (N-Dekker) [inline azure-pipelines.yml]: Would it be possible "share" this git-hash with the other sections, `job: Windows` and `job: Linux`? (No show-stopper.)

Comment (kaspermarstal) [reply azure-pipelines.yml]: Yes we can make a variable at the top (like "PreferredToolArchitecture") and reference this variable in all sections. Would like to give it a go?

Comment (N-Dekker) [reply azure-pipelines.yml]: If azure is still running, I would first like to see it succeed. A "git hash" variable would certainly be nice, but it's no show-stopper to me, for the initial version.

Comment (kaspermarstal) [reply azure-pipelines.yml]: So far the failures seems to be related to the multi-threader changes. You have fixed this in another branch, right? If so, we can see on the pr-merge build if the issues are fully resolved before merging this PR.

Comment (kaspermarstal) [reply azure-pipelines.yml]: Oh, I see these changes were already merged. I can't figure out why some ITKv4 builds succeed while others don't, and some ITKv5 builds succeed and others don't. Can you take a look at the error messages azure?

Comment (N-Dekker) [reply azure-pipelines.yml]: Indeed I did merge #104 to develop, this morning. I had it open for review fo 23 hours, and I thought it wasn't necessary to wait any longer. The commits can easily be reverted later, if necessary. So the develop branch should now compile for both ITK4 and ITK5.

Comment (N-Dekker) [reply azure-pipelines.yml]: @kaspermarstal Linux ITKv5 says at https://dev.azure.com/kaspermarstal/Elastix/_build/results?buildId=127:

    Common/ImageSamplers/itkImageToVectorContainerFilter.hxx:278:35: error: ‘MultiThreader’ has not been declared

https://github.com/SuperElastix/elastix/blob/develop/Common/ImageSamplers/itkImageToVectorContainerFilter.hxx#L278	
	
Can you please try to fix it? It should be sufficient to just add `#include "itkMultiThreader.h"` to the elastix file "itkImageToVectorContainerFilter.hxx".

Comment (kaspermarstal) [reply azure-pipelines.yml]: Done, more progress, and we now get this: 

```
/home/vsts/work/1/s/Components/Metrics/AdvancedKappaStatistic/itkAdvancedKappaStatisticImageToImageMetric.hxx:673:86: error: there are no arguments to ‘vcl_ceil’ that depend on a template parameter, so a declaration of ‘vcl_ceil’ must be available [-fpermissive]
vcl_ceil( static_cast< double >( numPar ) / static_cast< double >( nrOfThreads ) ) );
```

Have you seen this before?

**Comment by kaspermarstal** [comment]
Merging, failures are not related to CI script, but to ITKv5 compatibility..

**Pull request merged**